examples/sir_hmc.py

```diff
@@ -242,35 +242,64 @@ def hook_fn(kernel, *unused):
 #
 # We first define a helper to create enumerated Categorical sites.
 
-def quantize(name, x_real, min, max):
-    """Randomly quantize in a way that preserves probability mass."""
+def quantize(name, x_real, min, max, spline_order=3):
+    """
+    Randomly quantize in a way that preserves probability mass.
+    We use a piecewise polynomial spline of order 3 or 5.
+    """
+    assert spline_order in [3, 5]
     assert min < max
     lb = x_real.detach().floor()
 
-    # This cubic spline interpolates over the nearest four integers, ensuring
-    # piecewise quadratic gradients.
     s = x_real - lb
     ss = s * s
     t = 1 - s
     tt = t * t
-    probs = torch.stack([
-        t * tt,
-        4 + ss * (3 * s - 6),
-        4 + tt * (3 * t - 6),
-        s * ss,
-    ], dim=-1) * (1/6)
-    q = pyro.sample("Q_" + name, dist.Categorical(probs)).type_as(x_real) - 1
-
-    x = lb + q
+
+    if spline_order == 3:
+        # This cubic spline interpolates over the nearest four integers, ensuring
+        # piecewise quadratic gradients.
+        probs = torch.stack([
+            t * tt,
+            4 + ss * (3 * s - 6),
+            4 + tt * (3 * t - 6),
+            s * ss,
+        ], dim=-1) * (1/6)
+    elif spline_order == 5:
+        # This quintic spline interpolates over the nearest eight integers, ensuring
+        # piecewise quartic gradients.
+        sss = ss * s
+        ssss = ss * ss
+        sssss = sss * ss
+
+        ttt = tt * t
+        tttt = tt * tt
+        ttttt = ttt * tt
+
+        probs = torch.stack([
+            2 * ttttt,
+            2 + 10 * t + 20 * tt + 20 * ttt + 10 * tttt - 7 * ttttt,
+            55 + 115 * t + 70 * tt - 9 * ttt - 25 * tttt + 7 * ttttt,
+            302 - 100 * ss + 10 * ssss,
+            302 - 100 * tt + 10 * tttt,
+            55 + 115 * s + 70 * ss - 9 * sss - 25 * ssss + 7 * sssss,
+            2 + 10 * s + 20 * ss + 20 * sss + 10 * ssss - 7 * sssss,
+            2 * sssss
+        ], dim=-1) * (1/840)
+
+    q = pyro.sample("Q_" + name, dist.Categorical(probs)).type_as(x_real)
+
+    x = lb + q - 1 if spline_order == 3 else lb + q - 3
     x = torch.max(x, 2 * min - 1 - x)
     x = torch.min(x, 2 * max + 1 - x)
+
     return pyro.deterministic(name, x)
 
 
 # Now we can define another equivalent model.
 
 @config_enumerate
-def continuous_model(data, population):
+def continuous_model(data, population, args):
```

Comment:
nit: Since this is user facing, could you please clean up so that all models have the same signature and that signature is `(data, args)`, replacing `population` by `args.population` below?

Reply:
done

Reply:
done please check piping in case i missed anything